src/public/Convert-SentinelARArmToYaml.ps1

```diff
@@ -140,106 +184,158 @@ function Convert-SentinelARArmToYaml {
             "query"
         )
 
+        # Use pipeline data and create a variable containing all parsed strings
+        if ($PsCmdlet.ParameterSetName -eq "Pipeline") {
+            $FullARM += $Data
+        }
+
         # Use parsed pipeline data if no file was specified (default)
         try {
             if ($PsCmdlet.ParameterSetName -eq "Pipeline") {
-                $AnalyticsRuleTemplate = $FullARM | ConvertFrom-Json -Verbose
+                $AnalyticsRuleTemplate = $FullARM | ConvertFrom-Json -Depth 99 -Verbose
             } else {
                 Write-Verbose "Read file `"$Filename`""
-                $AnalyticsRuleTemplate = Get-Content $Filename | ConvertFrom-Json -Verbose
+                $AnalyticsRuleTemplate = Get-Content $Filename | ConvertFrom-Json -Depth 99 -Verbose
             }
         } catch {
             throw "Could not convert source file. JSON might be corrupted"
         }
 
-        if ($AnalyticsRuleTemplate.resources.Count -ne 1) {
-            throw "ARM template must contain exactly one resource"
+        try {
+            if ((-not $AnalyticsRuleTemplate.resources) -or (($AnalyticsRuleTemplate.resources).Count -lt 1)) {
+                throw "This template contains no Analytics Rules"
+            }
+        } catch {
+            $PSCmdlet.ThrowTerminatingError($_)
         }
 
-        # Get the id of the analytic rule
-        if ($AnalyticsRuleTemplate.resources.id -match "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}") {
-            $Id = $Matches[0]
-        } else {
-            Write-Warning "Error reading current Id. Generating new Id."
-            $Id = (New-Guid).Guid
+        try {
```

Comment:
Line 211 to 219 can be removed as the logic is broken, I will post the fix a bit further down in the code

![image](https://github.com/f-bader/SentinelARConverter/assets/3834359/e9792052-14c0-4857-8282-7b711c2fc495)